### PR TITLE
drivers/periph/uart: add periph_uart_reconfigure feature & implementation for sam0

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This adds a function `uart_deinit_pins()` to turn off the SPI bus and free the GPIOs from SPI function so they can be used for something else. It also adds a `uart_init_pins()` function analogous to [`spi_init_pins()`](https://doc.riot-os.org/group__drivers__periph__spi.html#ga9fcf9ffd48cdc62c6019a2f07aab4ae9).

### Testing procedure

Nothing to test yet.

### Issues/PRs references

same as #13421 and #13631

### CODE REVIEW

File: cpu/sam0_common/periph/uart.c [hunk 1]
```diff
@@ -160,6 +168,40 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     return UART_OK;
 }
 
+void uart_init_pins(uart_t uart)
+{
+    init_pins(uart);
```
Comment (fjmolinas) [inline cpu/sam0_common/periph/uart.c:173]:
```suggestion
    _init_pins(uart);
```

Comment (benpicco) [reply cpu/sam0_common/periph/uart.c:173]: I think I'll just cherry-pick b4ea2ab4dff5553f63c66f3a17b08a0d0aa818a7 from #14326

Comment (fjmolinas) [reply cpu/sam0_common/periph/uart.c:173]: Ok!

**Comment by fjmolinas** [comment]
@benpicco this just needs a push to trigger actions!

**Pull request merged**